### PR TITLE
Support JDK 11 and JDK 17 Compilation

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Update the repository from mockito 1 to mockito 2 to support JDK 11 and JDK 17 compilation.

### When should this PR be merged
After this PR is merged to the main branch, we can't build the branch on JDK 8

### Version checked,

- openjdk "11.0.16.1"
- openjdk "17.0.4.1"

### Related Issue
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java [hunk 1]
```diff
@@ -457,10 +458,14 @@ public void testListUsersWithGET(List<org.wso2.carbon.user.core.common.User> use
 
         when(mockedUserStoreManager.getUserListWithID("http://wso2.org/claims/userid", "*", null)).thenReturn(users);
         when(mockedUserStoreManager.getRoleListOfUserWithID(anyString())).thenReturn(new ArrayList<>());
-        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
-        when(mockedGroupDAO.listSCIMGroups()).thenReturn(anySet());
-        when(mockedUserStoreManager.getSecondaryUserStoreManager("PRIMARY")).thenReturn(mockedUserStoreManager);
+
+        //whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java:462]: Why we are commenting these lines?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java [hunk 2]
```diff
@@ -636,20 +646,27 @@ public void testFilteringUsersWithGETWithPagination(List<org.wso2.carbon.user.co
...
+                anyInt(), nullable(String.class), nullable(String.class))).thenReturn(filteredUsersWithoutPagination);
 
         when(mockedUserStoreManager.getUserListWithID(any(Condition.class), anyString(), anyString(), eq(users.size()),
-                anyInt(), anyString(), anyString())).thenReturn(filteredUsersWithoutPagination);
+                anyInt(), nullable(String.class), nullable(String.class))).thenReturn(filteredUsersWithoutPagination);
 
         when(mockedUserStoreManager.getRoleListOfUserWithID(anyString())).thenReturn(new ArrayList<>());
-        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
-        when(mockedGroupDAO.listSCIMGroups(anyInt())).thenReturn(anySet());
+        if(domain=="PRIMARY"){
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java:659]: Can we optimise this using one if logic

File: pom.xml [hunk 3]
```diff
@@ -44,6 +44,11 @@
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>com.sun.xml.parsers</groupId>
```
Comment (Thumimku) [inline pom.xml:48]: Is this a new dependency? why do we need this dependency?

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 4]
```diff
@@ -30,6 +30,11 @@
     <packaging>bundle</packaging>
 
     <dependencies>
+        <dependency>
+            <groupId>com.sun.xml.parsers</groupId>
+            <artifactId>jaxp-ri</artifactId>
+            <version>1.4.5</version>
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml:36]: Can we remove the version?

File: pom.xml (hunk 3 above)
Comment (Thumimku) [inline pom.xml:48]: Please check the license version and vulnerability if we are adding this to our IS pack?

File: pom.xml [hunk 5]
```diff
@@ -44,6 +44,12 @@
...
+                <artifactId>jaxp-ri</artifactId>
```
Comment (piraveena) [inline pom.xml:49]: check whether it is added to the pack

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 6]
```diff
@@ -229,6 +237,20 @@
                     <systemPropertyVariables>
                         <jacoco-agent.destfile>target/jacoco.exec</jacoco-agent.destfile>
                     </systemPropertyVariables>
+                    <argLine>
+                        --add-opens java.base/jdk.internal.loader=ALL-UNNAMED
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml:241]: did we check whether this is fixed?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java [hunk 7]
```diff
@@ -638,18 +638,21 @@ public void testFilteringUsersWithGETWithPagination(List<org.wso2.carbon.user.co
...
```
Comment (Achintha444) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java:651]: Formating Issue.

File: features/org.wso2.carbon.identity.scim2.provider.feature/pom.xml [hunk 8]
```diff
@@ -22,7 +22,7 @@
         <groupId>org.wso2.carbon.identity.inbound.provisioning.scim2</groupId>
         <artifactId>identity-inbound-provisioning-scim2</artifactId>
         <relativePath>../../pom.xml</relativePath>
-        <version>3.2.0-SNAPSHOT</version>
+        <version>3.4.0-SNAPSHOT</version>
```
Comment (piraveena) [inline features/org.wso2.carbon.identity.scim2.provider.feature/pom.xml:25]: it should be 3.3.0-SNAPSHOT

**Review by piraveena — APPROVED** [review]

**Pull request merged**